### PR TITLE
[SPARK-47227][DOCS] Improve documentation for Spark Connect

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
Small update on explaining what is changing with Spark Connect and the clients to be more explicit about the changes.

image

### Why are the changes needed?
Doc

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
N/A

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Review by hvanhovell — APPROVED** [review]
LGTM

File: docs/spark-connect-overview.md [hunk 1]
```diff
@@ -56,6 +56,26 @@ client through gRPC as Apache Arrow-encoded row batches.
   <img src="img/spark-connect-communication.png" title="Spark Connect communication" alt="Spark Connect communication" />
 </p>
 
+## What is changing with Spark Connect
```
Comment (nchammas) [inline docs/spark-connect-overview.md:59]: This heading makes it sound like Spark Connect used to work differently and is now changing. That is not the case, is it?

I think a better heading would be something along the lines of: "How Spark Connect differs from "traditional" Spark programs"

I couldn't find a better word to describe non-Connect programs than "traditional", but you get the gist.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: docs/spark-connect-overview.md [hunk 2]
```diff
@@ -56,6 +56,26 @@ client through gRPC as Apache Arrow-encoded row batches.
...
+isolation of the client from the server. As a consequence, there are some changes
+that developers need to be aware of when using Spark Connect:
+
+1. The client does not run in the same process as the Spark driver. This means that
+   the client cannot directly access and interact with the driver JVM to manipulate
+   the execution environment. In particular, in PySpark, the client does not use Py4J
+   and thus the accessing the private fields holding the JVM implementation of `DataFrame`,
+   `Column`, `SparkSession`, etc. is not possible (e.g. `df._jdf`).
+2. By design, the Spark Connect protocol is designed around the concepts of Sparks logical
+   plans as the abstraction to be able to declarative describe the operations to be executed
```
Comment (nchammas) [inline docs/spark-connect-overview.md:71]: Nit: Avoid repeating "design"; tweak some grammar.

```suggestion
2. By design, the Spark Connect protocol uses Spark logical
   plans as the abstraction to declaratively describe the operations to be executed
```

**Review by nchammas — COMMENTED** [review]
Nice improvement! It answers the most important questions a user might ask about Connect vs. non-Connect programs.

> Does this PR introduce any user-facing change?
>
> No

Technically yes, since this is user-facing documentation. 😄

**Comment by grundprinzip** [comment]
@nchammas thanks for the feedback! Will address in a follow up!

**Pull request closed without merging**